redlines/cli.py

```
@@ -85,87 +193,437 @@ def print_intro():
+            try:
+                from .pdf import PDF_AVAILABLE, PDFFile
+
+                if not PDF_AVAILABLE:
+                    raise click.ClickException(
+                        f"PDF support requires the 'pdf' extra.\n\n"
+                        f"Cause: The pdfplumber package is required for PDF support but is not installed.\n\n"
+                        f"To fix: Install with PDF support:\n"
+                        f"  pip install redlines[pdf]\n\n"
+                        f"  # Or install pdfplumber directly\n"
+                        f"  pip install pdfplumber"
+                    )
```

Ruff (0.14.10)

Abstract `raise` to an inner function

(TRY301)